src/main/java/com/exactpro/epfast/decoder/integer/DecodeMandatoryInt32.java

```diff
@@ -36,10 +70,16 @@ public void continueDecode(ByteBuf buf) {
         int readerIndex = buf.readerIndex();
         int readLimit = buf.writerIndex();
         if (value >= 0) {
+            if (checkPositiveSingExtension) {
+                checkOverlongPositive(0, buf.getByte(readerIndex)); //continue checking
```

Comment:
reset check negative flag

Reply:
Have you tried that?

Reply:
there is no more check negative flag, I combined both as a single flag

Reply:
Ok. Reset check overlong flag here